rmc/utils/simultaneous_breaks.py

```diff
@@ -542,10 +542,14 @@ def process_section_group(
         # find the one "best" breakpoint (breakpoint with largest chi square value)
         if ht.count() > 0:
             ht = annotate_max_chisq_per_section(ht, freeze)
-            ht = ht.filter(ht.max_chisq == ht.section_max_chisq)
+            ht = ht.filter(ht.chisq == ht.section_max_chisq)
 
     ht = ht.annotate_globals(chisq_threshold=chisq_threshold)
-    ht = ht.naive_coalesce(output_n_partitions)
+    # NOTE: Changed `naive_coalesce` to `repartition` here to fix Hail bug:
+    # https://discuss.hail.is/t/zip-length-mismatch-error/3548/7
+    # TODO: Revert when bug has been resolved
```

Comment:
maybe we can also revert this now and add a note that switching to `repartition` could fix the zip length bug

Reply:
Done